### PR TITLE
Form accessibility fixes

### PR DESCRIPTION
**Pull request by davidrapson**
Ran the checkout pages through a validator and [tota11y](http://khan.github.io/tota11y/) and fixed a few accessibility issues. Largely around how we used fieldsets and how we associate fields.

- Reworks form markup to allow for the correct usage of `<fieldset>` and
`<legend>`.
- Fixes some field label assignments
- Corrects heading levels on checkout page
- Extracts fields out into fragments to make this template a little more
  readable, and easier to change.

Re-do of https://github.com/guardian/subscriptions-frontend/pull/177 which needed too many conflicts fixing.

@afiore @tudorraul

### CODE REVIEW

File: app/views/checkout/payment.scala.html [hunk 1]
```diff
@@ -14,168 +12,95 @@
...
+                    <div class="js-append">
+                        @fragments.checkout.notices()
+                    </div>
                 </div>
             </div>
-        </div>
 
-        <div class="checkout-container__form">
-
-            <form class="form js-checkout-form" method="POST">
```
Comment (tudorraul) [inline app/views/checkout/payment.scala.html]: was this a nested form?

Comment (davidrapson) [inline app/views/checkout/payment.scala.html]: Yup :smile:

Comment (davidrapson) [inline app/views/checkout/payment.scala.html]: Browsers are *really* resilient it turns out

File: app/views/checkout/payment.scala.html [hunk 2]
```diff
@@ -14,168 +12,95 @@
...
+                        <div class="field-panel__edit">
+                            <a href="#yourDetails" class="text-button u-button-reset js-edit-your-details">Edit</a>
                         </div>
+                        <div class="field-panel__fields">
+                            @fragments.checkout.fieldsPersonal(form, userIsSignedIn)
 
-                        <div class="actions">
-                            <a href="#paymentDetails" class="button button--primary button--large js-checkout-your-details-submit">Continue</a>
+                            <div class="actions">
+                                <a href="#paymentDetails" class="button button--primary button--large js-checkout-your-details-submit">Continue</a>
```
Comment (tudorraul) [inline app/views/checkout/payment.scala.html]: will `href` reference still work?

File: app/views/fragments/checkout/fieldsPayment.scala.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <input type="text" class="input-text input-text--small js-input"
+        name="payment.sortcode" id="payment-sortcode" maxlength="8"
+        placeholder="00-00-00" required
+        data-masker-delim="-" data-masker-length="2">
+    @fragments.forms.errorMessage("")
+</div>
+<div class="form-field js-checkout-holder">
+    <label class="label" for="payment-holder">Name of account holder</label>
+    <input type="text" maxlength="18" class="input-text js-input" id="payment-holder" name="payment.holder">
+    @* TODO: Why limited to 18 characters? *@
```
Comment (tudorraul) [inline app/views/fragments/checkout/fieldsPayment.scala.html]: cc @TesterSpike to the why question

File: app/views/checkout/payment.scala.html (hunk 2 above)
Comment (davidrapson) [inline app/views/checkout/payment.scala.html]: Yep, the `id` is still on the container below, l63 `<div id="paymentDetails" class="field-panel is-collapsed js-fieldset-payment-details">`

File: app/views/fragments/checkout/fieldsPayment.scala.html (hunk 3 above)
Comment (tudorraul) [inline app/views/fragments/checkout/fieldsPayment.scala.html]: I would remove this comment once answered

**Comment by tudorraul** [comment]
:+1: bar the todo comment

File: app/views/fragments/checkout/fieldsPayment.scala.html (hunk 3 above)
Comment (TesterSpike) [inline app/views/fragments/checkout/fieldsPayment.scala.html]: Apparently it's part of the BACS service limits http://www.bacs.co.uk/Bacs/Businesses/Resources/Pages/Glossary.aspx

"The payer’s account name (maximum of 18 characters). This must be the name of the person who is paying the Direct Debit and has signed the Direct Debit Instruction (DDI)."

**Comment by TesterSpike** [comment]
:+1: nothing appears to have regressed

**Comment by davidrapson** [comment]
"nothing appears to have regressed". That's all I ever aim for really.

**Comment by tudorraul** [comment]
@TesterSpike This might fix the intermittent issues with validating the email address, since the nested form was removed. Maybe!

**Pull request merged**